### DISCUSSION TITLE
Frequency of powered device polling- Can this be set? What is default?

### DISCUSSION THREAD

**Discussion by apella12** [zwave-js/zwave-js#6707] [Request Support / Investigate Issue] [answered]
For when you return; Using Aeotec HEM DSB28 as example. I have set up the reporting for only the whole HEM (EP 0) with watts based on thresholds and a daily KWH.  I have noticed via Zniffer a full series of GET/REPORT of what looks like all the meter items (including all endpoints and voltage and amps) periodically (seems like daily-ish). Is that a configurable frequency? I'm using ZUI.  As reference with the OpenHab ZW binding, power device polling could be set to 10 days.  It is not causing a problem, but is extra unnecessary radio traffic.

**Comment by AlCalzone** [comment, marked as answer]
The 6 hours are a specification requirement:
![grafik](https://github.com/zwave-js/node-zwave-js/assets/17641229/3fd1f33c-1abb-4ba1-8574-c5d469647794)

I'll have to check whether Z-Wave JS does not recognize here that unsolicited reports are received.

Can you provide me with logs of an interview and possibly a couple of hours of traffic, including an unsolicited report and a series of polls?

**Comment by apella12** [comment]
Let me think about this (new information). I have the house pretty fully metered and as you may recall, I'm probably too focused on minimizing zwave traffic. 

My devices are basically setup to report just watts and a midnight refresh to get KWH.  I guess the question is, if a watt report is received within 6 hours will that reset the clock for the whole Meter CC?  As an example, I did notice the default behavior on my A/C (aeotec HEM) meter that has 8 hour reporting during the winter. If I can change to 6 for just watts and that resets the clock, I can make that work.

**Reply by AlCalzone** [reply]
You can try that. I'll have to look at the code again if there's an heuristic or if Z-Wave JS just checks if there was an unsolicited meter report in the last 6h.

**Comment by apella12** [comment]
I'm starting to plow through the event logs node by node (so far just two Aeotec DSB09).  I believe the 6 hrs. is enforced by endpoint. I have the coils on two separate circuits (EP 1 and EP2) with threshold reporting.  I have disabled the "whole HEM" readings as meaningless. One of the circuits stays within the threshold for over 6 hours, so gets polled (watt/KWH) and also the whole HEM is polled (EP 0 watt/KWH) at that time. I'd say that ZW specification negates both threshold settings (to reduce traffic) and the disable/enable (to eliminate it-as not needed by the user). I can kind of understand 6 hrs. as a minimum on a threshold, but don't think disabled meter parameters should be polled.  That is why they are disabled. I have trouble seeing the logic here.
OpenHab kind of has this setup, a channel (like a voltage report) that isn't linked to an item (items are persisted and used in graphs, etc) are not polled by the binding because they aren't needed by the user.
Also, on a related note I have the multisensor (49) power readings disabled and those are also polled. 

More to come as time permits.

**Comment by apella12** [comment]
I question the logic of the 6 hours since it overrides user settings (such as Thresholds and disables), but you can't fight city hall. I added a few unnecessary unsolicited reports to mostly eliminate even more unneeded information and the polling traffic itself. The only enhancement suggestion would be to have the clock reset by device, not endpoint.  I could add that as issue, if you think it would be okay? Otherwise I'm ok.

**Reply by AlCalzone** [reply]
> The only enhancement suggestion would be to have the clock reset by device, not endpoint.

I think this needs to be a per-device compat setting, since there are devices that only report automatically on some of their endpoints, and those would be broken by such a change.
And since this will end up being a compat flag, I think simply making that flag disable the timed polls would be easiest to work with.

**Comment by apella12** [comment]
Good idea !  That will work for me.
